### PR TITLE
[Bugfix] Fix `PerTensorScaleParameter` weight loading for fused models

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Small update to fix weight loading of `PerTensorScaleParameter` and `BasevLLMParameter`  for fused models

### CODE REVIEW

**Review by mgoin — APPROVED** [review]

File: vllm/model_executor/layers/linear.py [hunk 1]
```diff
@@ -573,11 +574,13 @@ def weight_loader_v2(self,
...
-        param_data = param.data
         if loaded_shard_id is None:
-            if param.output_dim is None:
-                assert param_data.shape == loaded_weight.shape
-                param_data.copy_(loaded_weight)
+            if isinstance(param, PerTensorScaleParameter):
+                param.load_merged_column_weight(loaded_weight=loaded_weight,
+                                                shard_id=0)
+                return
+            elif type(param) is BasevLLMParameter:
```
Comment (mgoin) [inline vllm/model_executor/layers/linear.py:582]: switch to use isinstance

File: vllm/model_executor/layers/linear.py [hunk 2]
```diff
@@ -720,11 +723,13 @@ def weight_loader_v2(self,
...
-        param_data = param.data
         if loaded_shard_id is None:  # special case for certain models
-            if param.output_dim is None:
-                assert param_data.shape == loaded_weight.shape
-                param_data.copy_(loaded_weight)
+            if isinstance(param, PerTensorScaleParameter):
+                param.load_merged_column_weight(loaded_weight=loaded_weight,
+                                                shard_id=0)
+                return
+            elif type(param) is BasevLLMParameter:
```
Comment (mgoin) [inline vllm/model_executor/layers/linear.py:731]: ditto

File: vllm/model_executor/layers/linear.py (hunk 1 above)
Comment (dsikka) [reply vllm/model_executor/layers/linear.py:582]: we dont want `isinstance` in this case. all the parameters we introduced as instances of `BasevLLMParameter` so they'll all return True for `instance`. Specifically targeting parameters of the type `BasevLLMParameter`

File: vllm/model_executor/layers/linear.py (hunk 2 above)
Comment (dsikka) [reply vllm/model_executor/layers/linear.py:731]: same as above

File: vllm/model_executor/layers/linear.py (hunk 1 above)
Comment (mgoin) [reply vllm/model_executor/layers/linear.py:582]: I see, okay

**Pull request merged**